crates/flux-desugar/src/desugar.rs

```diff
@@ -1329,11 +1329,74 @@ trait DesugarCtxt<'genv, 'tcx: 'genv> {
                     self.genv().alloc(e2?),
                 )
             }
+            surface::ExprKind::Constructor(path, constructor_args) => {
+                let path = self.desugar_constructor_path(path)?;
+                let field_exprs = constructor_args
+                    .iter()
+                    .filter_map(|arg| {
+                        match arg {
+                            ConstructorArgs::FieldExpr(e) => Some(e),
+                            ConstructorArgs::Spread(_) => None,
+                        }
+                    })
+                    .collect::<Vec<&FieldExpr>>();
+
+                let field_exprs = try_alloc_slice!(self.genv(), field_exprs, |field_expr| {
+                    let e = self.desugar_expr(&field_expr.expr)?;
+                    Ok(fhir::FieldExpr {
+                        ident: field_expr.ident,
+                        expr: e,
+                        fhir_id: self.next_fhir_id(),
+                        span: e.span,
+                    })
+                })?;
+
+                let spreads = constructor_args
+                    .iter()
+                    .filter_map(|arg| {
+                        match arg {
+                            ConstructorArgs::FieldExpr(_) => None,
+                            ConstructorArgs::Spread(s) => Some(s),
+                        }
+                    })
+                    .collect::<Vec<&Spread>>();
+
+                let spread = if spreads.len() == 0 {
+                    None
+                } else if spreads.len() == 1 {
+                    let s = spreads[0];
+                    let spread = fhir::Spread {
+                        expr: self.desugar_expr(&s.expr)?,
+                        span: s.span,
+                        fhir_id: self.next_fhir_id(),
+                    };
+                    Some(self.genv().alloc(spread))
+                } else {
+                    // Multiple spreads found - emit and error
+                    return Err(self.emit_err(errors::MultipleSpreadsInConstructor::new(
+                        spreads[1].span,
+                        spreads[0].span,
+                    )));
+                };
```

Comment:
maybe

```suggestion
                let spread = match &spreads[..] {
                    [] => None.
                    [s] => {
                        let spread = fhir::Spread {
                            expr: self.desugar_expr(&s.expr)?,
                            span: s.span,
                            fhir_id: self.next_fhir_id(),
                        };
                        Some(self.genv().alloc(spread))
                    }
                    [s1, s2, ..] => {
                        // Multiple spreads found - emit and error
                         return Err(self.emit_err(errors::MultipleSpreadsInConstructor::new(
                              s1.span,
                              s2.span,
                          )));
                    }
                };
```

Reply:
Oh cool I didn't know you could do that actually